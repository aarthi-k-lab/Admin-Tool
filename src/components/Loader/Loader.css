.sign-in-loader {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.message {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-size: 1.25rem;
}
