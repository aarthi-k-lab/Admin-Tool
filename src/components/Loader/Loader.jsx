import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import Center from 'components/Center';
import PropTypes from 'prop-types';
import './Loader.css';

const Loader = ({ message }) => (
  <Center>
    <span styleName="message">
      <CircularProgress size={50} />
      {message}
    </span>
  </Center>
);

Loader.defaultProps = {
  message: '',
};
Loader.propTypes = {
  message: PropTypes.string,
};
export default Loader;
