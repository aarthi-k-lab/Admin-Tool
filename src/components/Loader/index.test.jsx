import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '.';

describe('<Loader />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<Loader>content</Loader>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
