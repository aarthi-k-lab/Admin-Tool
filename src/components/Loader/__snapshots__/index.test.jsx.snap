// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> snapshot test 1`] = `
<div
  className="components-Center-Center__center--expand--1OMMm components-Center-Center__center--QKyYo"
>
  <span
    className="components-Loader-Loader__message--fhm4A"
  >
    <div
      className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
      role="progressbar"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-6"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    
  </span>
</div>
`;
